src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -122,34 +130,57 @@ public S3Storage(final S3AsyncClient client, final String bucket) {
 
     @Override
     public CompletableFuture<Void> save(final Key key, final Content content) {
-        return this.client.createMultipartUpload(
-            CreateMultipartUploadRequest.builder()
-                .bucket(this.bucket)
-                .key(key.string())
-                .build()
-        ).thenApply(
-            created -> new MultipartUpload(
-                new Bucket(this.client, this.bucket),
-                key,
-                created.uploadId()
-            )
-        ).thenCompose(
-            upload -> upload.upload(content).handle(
-                (ignored, throwable) -> {
-                    final CompletionStage<Void> finished;
-                    if (throwable == null) {
-                        finished = upload.complete();
-                    } else {
-                        final CompletableFuture<Void> promise = new CompletableFuture<>();
-                        finished = promise;
-                        upload.abort().whenComplete(
-                            (result, ex) -> promise.completeExceptionally(throwable)
-                        );
+        final CompletableFuture<Void> future;
+        final Optional<Long> size = content.size();
+        if (size.isPresent() && size.get() < S3Storage.MIN_MULTIPART) {
+            future = this.client.putObject(
+                PutObjectRequest.builder()
+                    .bucket(this.bucket)
+                    .key(key.string())
+                    .build(),
+                new AsyncRequestBody() {
```

Comment:
@olegmoz let's extract this to a private static class to make it more readable. The reader don't really care about the inner structure of async request so let's hide it. We can have something like `ContentBody`.

Reply:
@Vatavuk created a private class for that, think it's event better. Please review